Gems/EMotionFX/Code/EMotionFX/Rendering/Common/RotateManipulator.cpp

```diff
@@ -367,6 +371,77 @@ namespace MCommon
         }
     }
 
+    bool RotateManipulator::internalCheckIntersectXAxis(
+        const MCommon::Camera::ProjectionMode projectionMode,
+        const MCore::Ray& ray,
+        const AZ::Vector3& camRollAxis,
+        AZ::Vector3& normalRotationDirection)
+    {
+        // internal check for rotation around x axis, if the following intersection conditions are fulfilled
+        // innerAABB not hit, outerAABB hit, innerBoundingSphere hit and angle between cameraRollAxis and clickPosition > pi/2
+        const AZ::Vector3 rayDirectionRCP = ray.GetDirection().GetReciprocal();
+        if ((AZ::Intersect::IntersectRayAABB2(ray.GetOrigin(), rayDirectionRCP, m_xAxisInnerAabb) == AZ::Intersect::ISECT_RAY_AABB_NONE ||
+             (projectionMode == MCommon::Camera::PROJMODE_ORTHOGRAPHIC && m_xAxisVisible)) &&
+            ray.Intersects(m_innerBoundingSphere))
+        {
+            float start = 0, end = 0;
+            if (AZ::Intersect::IntersectRayAABB2(ray.GetOrigin(), rayDirectionRCP, m_xAxisAabb, start, end) !=
+                AZ::Intersect::ISECT_RAY_AABB_NONE)
+            {
+                normalRotationDirection = ((ray.GetOrigin() + ray.GetDirection() * start) - m_position).GetNormalized();
+                return MCore::Math::ACos(camRollAxis.Dot(normalRotationDirection)) > MCore::Math::halfPi;
+            }
+        }
+        return false;
```

Comment:
update: It looks like there is a large amount of repeated code across each axis, can we not parameterize the function to take the axis as an input and save all this duplication? 🤔 Just seems like a shame to have so much duplication when there is maybe only a small piece that's different..

I did a quick diff between the code and it looks like we can just pass in `m_<xyz>AxisInnerAabb`, `m_<xyz>AxisVisible` and `m_<xyz>AxisAabb` (make an helper function do to this that's not part of the public API and have the existing functions call into that)